### ISSUE TITLE
Instantly crashing upon opening

### ISSUE THREAD

**Issue by L31FY** [Tobi823/ffupdater#434]
**Describe the bug**
Cannot open the app due to instant crash. 

**To Reproduce**
1. Attempt to open app.
2. App crashes with error instantly. 

**Expected behavior**
App opens

**Error message from FFUpdater**
Stacktrace:
```
java.lang.NullPointerException: Attempt to invoke virtual method 'com.google.gson.JsonObject com.google.gson.JsonElement.getAsJsonObject()' on a null object reference
	at de.marmaro.krt.ffupdater.network.mozillaci.MozillaCiJsonConsumer.findChainOfTrustJson(Unknown Source:172)
	at de.marmaro.krt.ffupdater.network.mozillaci.MozillaCiJsonConsumer$findChainOfTrustJson$1.invokeSuspend(Unknown Source:12)
	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(Unknown Source:11)
	at kotlinx.coroutines.DispatchedTask.run(Unknown Source:88)
	at android.os.Handler.handleCallback(Handler.java:942)
	at android.os.Handler.dispatchMessage(Handler.java:99)
	at android.os.Looper.loopOnce(Looper.java:226)
	at android.os.Looper.loop(Looper.java:313)
	at android.app.ActivityThread.main(ActivityThread.java:8757)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:571)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1067)
	Suppressed: kotlinx.coroutines.DiagnosticCoroutineContextException: [StandaloneCoroutine{Cancelling}@faebcac, Dispatchers.Main]
Timestamp: 2023-07-12T08:24:27.519

```
Device information:
| Key | Value |
| --- | --- |
| FFUpdater version | 78.0.1 (143) release |
| Device | SM-G998U1 (p3quew, p3q, lahaina) |
| Manufacturer | samsung (samsung) | 
| Supported ABIs | arm64-v8a, armeabi-v7a, armeabi |
| Android version | 13 (SDK: 33) |
| OS | 21DJFA21, dpi, release-keys, 1684837107000 |

**Comment by bilemke** [comment]
I had to clear data for the app to get our if the crash loop on update from the prior release.

**Comment by Tobi823** [comment]
But it works now?

**Comment by bilemke** [comment]
> But the app works now?

For me, yes.

I may post a different issue though that may or may not be related. The crash occurring didn't happen on my android 12 device. Just my newer 13 device. On the 13 device, after I got it to open, it shows two instances of Frefox Nightly in the list.

**Comment by Tobi823** [comment]
> I may post a different issue though that may or may not be related. The crash occurring didn't happen on my android 12 device. Just my newer 13 device. On the 13 device, after I got it to open, it shows two instances of Frefox Nightly in the list.

I will look into but I don't think that the previous error is related to this bug. I changed the program logic for update checking and forgot to handle all cases (like an invalid cache response)

**Comment by Tobi823** [comment]
I will improve the handling of invalid API responses

**Comment by L31FY** [comment]
I do notice that there are two of them but I admit I was not paying attention enough to realize before it was mentioned. Uninstall and fresh install of the newer version cured the crash loop and there's not exactly any settings I can't replace in seconds so not a pain.

**Comment by Tobi823** [comment]
Both issues (crashes + Firefox Nightly twice) are fixed in 78.0.3
